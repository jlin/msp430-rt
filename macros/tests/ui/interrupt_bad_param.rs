#![no_main]

use msp430_rt_macros::{entry, interrupt};

#[entry]
fn main() -> ! {
    loop {}
}

#[interrupt]
fn TIM2(i: u32) -> ! {
    loop {}
}
