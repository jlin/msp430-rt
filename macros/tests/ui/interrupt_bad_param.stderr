error: `#[interrupt]` handlers must have signature `[unsafe] fn([<name>: CriticalSection]) [-> !]`
  --> tests/ui/interrupt_bad_param.rs:11:1
   |
11 | fn TIM2(i: u32) -> ! {
   | ^^^^^^^^^^^^^^^^^^^^
